File: 프로그래머스/src/프로그래머스_02262022/제일_작은_수_제거하기.cpp
#include <string>
#include <vector>
using namespace std;

vector<int> solution(vector<int> arr)
{
    vector<int> answer;
    int min = -1;
    if (arr.size() == 1)
        return {-1};
    //간단하니까 algorithm header 없이 진행
    for (auto i : arr)
        min = min < i && min > 0 ? min : i;
    for (auto i : arr)
        if (i != min)
            answer.emplace_back(i);
    return answer;
}

File: 2026/Level1/제일 작은 수 제거하기.py
# https://school.programmers.co.kr/learn/courses/30/lessons/12935?language=python3
def solution(arr):
    answer = []
    if len(arr) <= 1 :
        answer.append(-1)
    else :
        minValue = min(arr)
        for x in arr :
            if x != minValue :
                answer.append(x)    
    return answer